perllib/FixMyStreet/App/Controller/Waste.pm

```diff
@@ -390,6 +391,34 @@ sub populate_dd_details : Private {
     $c->cobrand->call_hook( 'garden_waste_dd_munge_form_details' => $c );
 
     $c->stash->{redirect} = $c->cobrand->call_hook( 'garden_waste_dd_redirect_url' => $p ) || '';
+
+    if ($c->cobrand->direct_debit_collection_method eq 'internal') {
+        my $form = FixMyStreet::App::Form::Waste::BankDetails->new(
+            c => $c,
+            page_name => 'bank_details',
+        );
+
+        if ($c->req->method eq 'POST') {
+            $form->process(params => $c->req->params);
+            if ($form->validated) {
+                my $success = $c->cobrand->call_hook('setup_direct_debit', [$form->value]);
+                if ($success) {
+                    $c->res->redirect($c->uri_for('/waste/dd_complete', {
+                        token => $c->stash->{reference},
+                        id => $c->stash->{report}->id
+                    }));
+                    $c->detach;
+                } else {
+                    $form->add_form_error('Failed to setup direct debit');
+                }
+            }
+        }
+
+        $c->stash->{form} = $form;
+    }
+
+    $c->stash->{template} = 'waste/dd.html';
+    $c->detach;
```

Comment:
I'm trying to display an extra form to gather the users details instead of redirecting them off to a portal like we do for other places, but it's not working as it should.

I think when the POST request is made for the Direct Debit details form it's picked up as a re-submission of the garden form, and so goes back to the page with the terms and conditions checkbox again, instead of submitting the Direct Debit.

I tried to alter the existing garden form to add a new page for the direct debit, but I couldn't get that working, I think in part because there are a lot of assumptions that the summary page will be the final page.

I must be missing a much easier way to do this?